### DISCUSSION TITLE
firejail download folder

### DISCUSSION THREAD

**Discussion by Rosika2** [netblue30/firejail#6090] [Q&A] [answered]
Hi all, :wave: 

I´m asking on behalf of a friend of mine, but I´m using my personal setup for reference here, as it is more convenient for me to follow any how-to´s.

Both my friend and I are using **Linux Lite**.

We want to use `google-chrome` in a firejailed environment, using the `--private` option together with a defined working directory. Here´s the respective command:

`firejail --private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/prov google-chrome`.

 In this case a new and empty home directory is mounted over the existing one and it has its own download folder, which differs from the system-wide download folder, right?

 Now let´s take the above mentioned scenario as a starting point: 

`firejail --private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/prov google-chrome`.

 But we need to have _access to the system-wide download folder as well_. 
That is: the firejailed instance of google-chrome should **see and have read and write access to its own download folder plus the system-wide download folder**.
Is this doable at all :question: 

I already copied the `google-chrome.profile` to `/home/rosika/.config/firejail` and added `whitelist ~/Downloads/` to it.
That alone wouldn´t work. So I need some help with that. :thinking: 

Many thanks for your help in advance and many greetings.

Rosika :slightly_smiling_face:

**Comment by rusty-snake** [comment, marked as answer]
You can bind mount the host's download dir to a second place and make this accessiable for the sandbox.

**Comment by kmk3** [comment]
Related discussion:

* <https://github.com/netblue30/firejail/discussions/4746>

**Comment by Rosika2** [comment]
Hi again, :wave: 

thanks to both of you for your replies.

@rusty-snake : 

That´s the solution I was looking for. It hadn´t occurred to me though.

Here´s what I did: 

- I created the folder 'system-wide_downloads' within the private sandbox. 
- `sudo mount --bind /home/rosika/Downloads /media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/prov/system-wide_downloads`
-  `firejail --private=/media/rosika/f14a27c2-0b49-4607-94ea-2e56bbf76fe1/DATEN-PARTITION/Dokumente/prov google-chrome`

 Both download folders are available within the private sandbox. 
As the system-wide download folder resides within firejail´s home directory it´s available without any further ado.

Thanks a lot. :heart: 

@kmk3 : 

Thanks to you  too, Kelvin, for providing the link. I´ll have a good look at it.

Many greetings to all

Rosika :slightly_smiling_face:
